lib/opentelemetry_datadog/config/parser.ex

```diff
@@ -0,0 +1,67 @@
+defmodule OpentelemetryDatadog.Config.Parser do
+  @moduledoc """
+  Environment variable parser and type converter for Datadog configuration.
+  """
+
+  @type validation_error :: {:error, :missing_required_config | :invalid_config, String.t()}
+
+  @doc "Unified environment variable getter with type conversion and validation."
+  @spec get_env(String.t(), atom(), keyword()) :: {:ok, any()} | validation_error()
```

Comment:
Usually errors are typed like:


```suggestion
  @spec get_env(String.t(), atom(), keyword()) :: {:ok, any()} | {:error, validation_error()}
```

where `validation_error()` is the union of errors.

Reply:
Fixed validation error type pattern